src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,203 @@
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with ID " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT car_id FROM cars_drivers WHERE driver_id = ?";
+        List<Long> carIds = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carIds.add(resultSet.getLong("car_id"));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get car IDs for driver ID "
+                    + driverId, throwable);
+        }
+        Optional<Car> carOptional;
+        List<Car> cars = new ArrayList<>();
+        for (Long id : carIds) {
+            carOptional = get(id);
+            carOptional.ifPresent(cars::add);
+        }
+        return cars;
+    }
+
+    private void assignDriversToCar(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                   PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't assign drivers to car " + car, throwable);
+        }
+    }
+
+    private Car parseCar(ResultSet resultSet) throws SQLException {
+        Long carId = resultSet.getLong("id");
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getLong("manufacturer_id");
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+        manufacturer.setId(manufacturerId);
+        Car car = new Car(model, manufacturer, Collections.emptyList());
+        car.setId(carId);
+        return car;
+    }
+
+    private List<Driver> getDriversFromDb(Car car) {
+        List<Driver> drivers = new ArrayList<>();
+        String query = "SELECT id, name, license_number FROM cars_drivers JOIN drivers "
+                + "ON cars_drivers.driver_id = drivers.id WHERE car_id = ? AND is_deleted = 0";
```

Comment:
we can use Aliases

Reply:
I think that creating an alias isn't worth it when I'm going to use it only once. 
Doesn't help readability